src/lib/containers/widgets/ImageButtonWithTooltip.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { FunctionComponent } from 'react'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import ReactTooltip from 'react-tooltip'
+import { TOOLTIP_DELAY_SHOW } from '../table/SynapseTableConstants'
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core'
+import { Dropdown } from 'react-bootstrap'
+
+type CustomImageProps = {
+  svgImg: string
+  altText: string
+}
+
+type ImageButtonWithTooltipProps = {
+  image: IconDefinition | CustomImageProps
+  callbackFn: Function
```

Comment:
I see in other parts of the PR that `callbackFn` is passed an empty function, e.g. () => null, these empty fns could instead be removed by making callbackFn as an optional prop and then if this PR pull's in the latest develop you could change the syntax below `onClick={() => callbackFn()}` to use optional chaining onClick={() => callbackFn?()}

Reply:
Thank you for pointing this out!
The logic is that it always needs a callbackFn unless it's a dropdown toggle.
So I'm getting rid of the toggle flag, making the callbackFn optional and use the logic of 
'if it doesn't have a callBack -- I assume it's a dropdown toggle'. Simplifies things a lot